storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManagerMultipleSubscriptionsTest.java

```diff
@@ -49,66 +51,34 @@
+    TopicBasedRemoteLogMetadataManagerMultipleSubscriptionsTest(ClusterInstance clusterInstance) {
+        this.clusterInstance = clusterInstance;
     }
 
-    @BeforeEach
-    public void setup() {
-        // Start the cluster only.
-        remoteLogMetadataManagerHarness.setUp(new EmptyTestInfo());
-    }
-
-    @AfterEach
-    public void teardown() throws IOException {
-        remoteLogMetadataManagerHarness.close();
-    }
-
-    @Test
+    @ClusterTest
     public void testMultiplePartitionSubscriptions() throws Exception {
         // Create topics.
         String leaderTopic = "leader";
-        HashMap<Object, Seq<Object>> assignedLeaderTopicReplicas = new HashMap<>();
-        List<Object> leaderTopicReplicas = new ArrayList<>();
         // Set broker id 0 as the first entry which is taken as the leader.
-        leaderTopicReplicas.add(0);
-        leaderTopicReplicas.add(1);
-        leaderTopicReplicas.add(2);
-        assignedLeaderTopicReplicas.put(0, JavaConverters.asScalaBuffer(leaderTopicReplicas));
-        remoteLogMetadataManagerHarness.createTopicWithAssignment(leaderTopic,
-            JavaConverters.mapAsScalaMap(assignedLeaderTopicReplicas),
-            remoteLogMetadataManagerHarness.listenerName());
+        createTopic(leaderTopic, Collections.singletonMap(0, Arrays.asList(0, 1, 2)));
 
         String followerTopic = "follower";
-        HashMap<Object, Seq<Object>> assignedFollowerTopicReplicas = new HashMap<>();
-        List<Object> followerTopicReplicas = new ArrayList<>();
         // Set broker id 1 as the first entry which is taken as the leader.
-        followerTopicReplicas.add(1);
-        followerTopicReplicas.add(2);
-        followerTopicReplicas.add(0);
-        assignedFollowerTopicReplicas.put(0, JavaConverters.asScalaBuffer(followerTopicReplicas));
-        remoteLogMetadataManagerHarness.createTopicWithAssignment(
-            followerTopic, JavaConverters.mapAsScalaMap(assignedFollowerTopicReplicas),
-            remoteLogMetadataManagerHarness.listenerName());
+        createTopic(followerTopic, Collections.singletonMap(0, Arrays.asList(1, 2, 0)));
 
         String topicWithNoMessages = "no-messages-topic";
-        HashMap<Object, Seq<Object>> assignedTopicReplicas = new HashMap<>();
-        List<Object> noMessagesTopicReplicas = new ArrayList<>();
         // Set broker id 1 as the first entry which is taken as the leader.
-        noMessagesTopicReplicas.add(1);
-        noMessagesTopicReplicas.add(2);
-        noMessagesTopicReplicas.add(0);
-        assignedTopicReplicas.put(0, JavaConverters.asScalaBuffer(noMessagesTopicReplicas));
-        remoteLogMetadataManagerHarness.createTopicWithAssignment(
-            topicWithNoMessages, JavaConverters.mapAsScalaMap(assignedTopicReplicas),
-            remoteLogMetadataManagerHarness.listenerName());
+        createTopic(topicWithNoMessages, Collections.singletonMap(0, Arrays.asList(1, 2, 0)));
```

Comment:
Could you please call `ClusterInstance#waitForTopic` to make sure all topics get created?

https://github.com/apache/kafka/blob/trunk/core/src/test/java/kafka/test/ClusterInstance.java#L185

Reply:
Updated it. Thanks for the suggestion.